apps/supernova/src/hooks/useBoundQuery.ts

```diff
@@ -3,20 +3,20 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { useQuery } from "@tanstack/react-query"
+import { useQuery, UseQueryResult } from "@tanstack/react-query"
 import { QUERY_FUNCTIONS } from "../api/queryFunctions"
 import { useGlobalsApiEndpoint } from "../components/StoreProvider"
 
-export const useBoundQuery = (key: keyof typeof QUERY_FUNCTIONS, { options }: any = {}) => {
+export const useBoundQuery = <T>(key: keyof typeof QUERY_FUNCTIONS, { options }: any = {}): UseQueryResult<T> => {
```

Comment:
What is the purpose here that the caller can specify the expected result type with `<T>`?